plugins/com.google.cloud.tools.eclipse.appengine.localserver/plugin.xml

```diff
@@ -100,4 +100,207 @@
+            behaviourClass="com.google.cloud.tools.eclipse.appengine.localserver.server.CloudSdkServerBehaviour"
+            class="com.google.cloud.tools.eclipse.appengine.localserver.server.CloudSdkServer"
+            description="%serverTypeDescription"
+            id="com.google.cloud.tools.eclipse.server.id"
+            initialState="stopped"
+            launchConfigId="com.google.cloud.tools.eclipse.CloudSdkLaunchConfigurationType"
+            launchModes="run,debug,profile"
+            name="%serverTypeName"
+            runtime="true"
+            runtimeTypeId="com.google.cloud.tools.eclipse.runtime.id1"
+            startBeforePublish="false"
+            supportsRemoteHosts="false">
+      </serverType>
+   </extension>
+   <extension
+         point="org.eclipse.debug.core.launchConfigurationTypes">
+      <launchConfigurationType
+            delegate="com.google.cloud.tools.eclipse.appengine.localserver.server.CloudSdkLaunchConfigurationDelegate"
+            id="com.google.cloud.tools.eclipse.CloudSdkLaunchConfigurationType"
+            modes="run, debug, profile"
+            name="%serverTypeName">
+      </launchConfigurationType>
+   </extension>
+   <extension
+         point="org.eclipse.debug.ui.launchConfigurationTypeImages">
+      <launchConfigurationTypeImage
+            configTypeID="com.google.cloud.tools.eclipse.CloudSdkLaunchConfigurationType"
+            icon="icons/gcp.png"
+            id="com.google.cloud.tools.eclipse.ExternalLaunchConfigurationTypeImage">
+      </launchConfigurationTypeImage>
+   </extension>
+   <extension
+         point="org.eclipse.debug.ui.launchConfigurationTabGroups">
+      <launchConfigurationTabGroup
+            class="com.google.cloud.tools.eclipse.appengine.localserver.ui.CloudSdkLaunchConfigurationTabGroup"
+            id="com.google.cloud.tools.eclipse.server.launchConfigurationTabGroup"
+            type="com.google.cloud.tools.eclipse.CloudSdkLaunchConfigurationType">
+      </launchConfigurationTabGroup>
+   </extension>
+   <extension
+         point="org.eclipse.wst.server.ui.serverImages">
+      <image
+            icon="icons/gcp.png"
+            id="com.google.cloud.tools.eclipse.image"
+            typeIds="com.google.cloud.tools.eclipse.runtime.id1,com.google.cloud.tools.eclipse.server.id">
+      </image>
+   </extension>
+   <extension
+         point="org.eclipse.wst.common.project.facet.ui.images">
+      <image
+            path="icons/gcp.png"
+            facet="com.google.cloud.tools.eclipse.facet">
+      </image>
+   </extension>
+   <extension
+         point="org.eclipse.ui.commands">
+      <command
+            id="com.google.cloud.tools.eclipse.deployCommand"
+            name="%cloudsdkdeploy">
+      </command>
+   </extension>
+   <extension
+         point="org.eclipse.core.expressions.propertyTesters">
+      <propertyTester
+            class="com.google.cloud.tools.eclipse.appengine.localserver.deploy.CloudSdkProjectPropertyTester"
```

Comment:
A few remarks:
- rather than define our own property tester, it's probably better from an extensibility perspective to provide an expression _definition_ (`org.eclipse.core.expressions.definition`) so we can vary the implementation
- WTP provides a [facet property tester](https://eclipse.org/articles/Article-BuildingProjectFacets/tutorial.html#property.tester)
- I think some of this is misnamed. CloudSDKProject seems to coarse grained, and I think we'll need to be able to distinguish between AE-Standard and AE-Flexible